internal/accounts/accounts_test.go

```diff
@@ -0,0 +1,131 @@
+/*
+ * Flow CLI
+ *
+ * Copyright 2019-2021 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package accounts
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/onflow/cadence"
+	"github.com/onflow/flow-go-sdk"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/onflow/flow-go-sdk/crypto"
+	"github.com/onflow/flow-go-sdk/test"
+)
+
+// One way to handle testing a result serializes as expected - convert it to a map[string]interface{} and make
+// many type assertions
+func TestAccountsResultJSON(t *testing.T) {
+
+	testCases := []struct {
+		Name    string
+		Account *flow.Account
+	}{
+		{Name: "Generated Account", Account: test.AccountGenerator().New()},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.Name, func(t *testing.T) {
+			account := tc.Account
+			result := AccountResult{
+				Account: tc.Account,
+				include: []string{"contracts"},
+			}
+			out := result.JSON()
+			EvaluateAccountJson(t, out, account)
```

Comment:
Although this is a good and detailed assertion I feel you could simplify it so you have less work. The way it could be implemented is to just `assert.Equal(t, result.JSON(), accountToJSON(account))` and have the function be:  `accountToJSON(account flow.Account) interace{}` where the logic can actually be copied from the internal layer (not reused) but copied, as we currently know this function behaves correctly, so this will detect any regression in that functionality, the same can go for `String()` and `Oneliner()`. This way you will save yourself quite some work and the main point of these tests is detecting regression. You can even just build an `interface{}` you know is correct and use that to test for equality.

Reply:
Sounds good - will go this way!